csharp/example/Program.cs

```diff
@@ -0,0 +1,134 @@
+﻿/*
+ * Copyright (C) 2023 Dremio Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Microsoft.Data.Analysis;
+using Grpc.Net.Client;
+using Grpc.Core;
+using Apache.Arrow.Flight.Client;
+using Apache.Arrow.Flight;
+using Apache.Arrow;
+
+namespace FlightClientExample
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            string host = "localhost";
+            string port = "32010";
+            string user = "dremio";
+            string pass = "dremio123";
+            // For default query, add the Samples source and Format the zips.json in Dremio UI
+            string query = "SELECT city, loc[0] AS longitude, loc[1] AS latitude, pop, state, _id FROM Samples.\"samples.dremio.com\".\"zips.json\" LIMIT 100";
+            string protocol = "http";
+
+            // Parse command-line arguments to override the defaults when set
+            for (int i = 0; i < args.Length-1; i++)
+            {
+                if (i % 2 == 0 && args[i].StartsWith("-")) 

```

Comment:
This logic assumes that the user will use the command flags correctly. I think a more robust algorithm would be to:
1. Have the method for parsing the command line arguments return a Dictionary <string, string> that has the keys being the command line flags and the values being the value of the flag.
2. When creating the Dictionary, have the initial values be the default values of the flags
3. Loop through the arguments list by each argument
4. Have a null string lastFlag to keep track of the previous command flag
5. Go through the following control structure for each argument:
```
// Skip if is null or white space
if (string.IsNullOrWhiteSpace(argument))
{
    continue;
}
// Add argument to lastFlag if starts with -
if (argument.StartsWith("-", StringComparison.Ordinal))
{
    lastFlag = argument;
}
// If lastFlag is not null, save the 
else if (lastFlag != null && optionsDic.ContainsKey(lastFlag))
{
    commandLineArguments.Add(lastFlag, argument);
    lastFlag = null;
}
```
6. Return the dictionary